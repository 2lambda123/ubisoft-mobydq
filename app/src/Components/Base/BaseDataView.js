import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { styles } from './../../styles/baseStyles';
import { withStyles } from '@material-ui/core/styles';
import { isSidebarOpen } from './../../actions/sidebar';

import CssBaseline from '@material-ui/core/CssBaseline';
import Content from './Content';
import { MessageBar } from './MessageBar';
import { setMessageBarOpen } from '../../actions/messageBar';
import { AppBarView } from './AppBar/AppBarView';
import { DrawerView } from './Drawer/DrawerView';
import AlertDialog from './AlertDialog';

class BaseDataView extends React.Component {
  setDrawerOpen = (setOpen) => {
    this.props.setSidebarOpen(setOpen);
  }

  render() {
    const { classes } = this.props;

    return (
      <React.Fragment>
        <CssBaseline />
        <div className={classes.root}>
          <AppBar
            position="absolute"
            className={classNames(classes.appBar, this.props.sidebarIsOpen && classes.appBarShift)}
          >
            <Toolbar disableGutters={!this.props.sidebarIsOpen} className={classes.toolbar}>
              <IconButton
                color="inherit"
                aria-label="Open drawer"
                onClick={() => this.setDrawerOpen(true)}
                className={classNames(
                  classes.menuButton,
                  'sidebarExpand',
                  this.props.sidebarIsOpen && classes.menuButtonHidden,
                )}
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="title" color="inherit" noWrap className={classes.title}>
                MobyDQ
              </Typography>
              {/*
              Removed bell icon because it wasn't used yet
              <IconButton color="inherit">
                <Badge badgeContent={4} color="secondary">
                  <NotificationsIcon />
                </Badge>
              </IconButton>
              */}
            </Toolbar>
          </AppBar>
          <Drawer
            variant="permanent"
            classes={{
              'paper': classNames(classes.drawerPaper, !this.props.sidebarIsOpen && classes.drawerPaperClose)
            }}
            open={this.props.sidebarIsOpen}
          >
            <div className={classes.toolbarIcon}>
              <IconButton onClick={() => this.setDrawerOpen(false)}>
                <ChevronLeftIcon />
              </IconButton>
            </div>
            <Divider />
            <List>{mainListItems}</List>
          </Drawer>
          <DrawerView classes={classes} sidebarIsOpen={this.props.sidebarIsOpen} setDrawerOpen={this.setDrawerOpen}/>
          <AlertDialog/>
          <main className={classes.content}>
            <Content />
          </main>
        </div>
        <MessageBar
          message={this.props.messageBarMessage}
          isOpen={this.props.messageBarIsOpen}
          setOpen={this.props.setMessageBarOpen}
        />
      </React.Fragment>
    );
  }
}

/*
 * BaseDataView.propTypes = {
 *   content: PropTypes.object.isRequired,
 * };
 */

const mapStateToProps = (state) => ({
  'sidebarIsOpen': state.sidebarIsOpen,
  'messageBarMessage': state.messageBarMessage
});

const mapDispatchToProps = (dispatch) => ({
  'setSidebarOpen': (sidebarOpenState) => dispatch(isSidebarOpen(sidebarOpenState)),
  'setMessageBarOpen': (open) => dispatch(setMessageBarOpen(open))
});

export default withStyles(styles)(withRouter(connect(mapStateToProps, mapDispatchToProps)(BaseDataView)));
