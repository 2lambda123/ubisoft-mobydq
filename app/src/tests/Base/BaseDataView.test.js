import React from 'react';
import { shallowWrap, mountWrap } from './../../setupTests';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';


import BaseDataView from './../../Components/Base/BaseDataView';

describe('BaseDataView component unit test', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWrap(
      <BaseDataView/>
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('BaseDataView component functional test', () => {
  let wrapper;
  const initialState = { 'sidebarIsOpen': false };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  it('renders the BaseDataView component with sidebar closed', () => {
    wrapper = mountWrap(<div><MemoryRouter><BaseDataView store={store}/></MemoryRouter></div>);
    expect(wrapper.find('AppBar').length === 1).toEqual(true);
    wrapper.find('IconButton').first().simulate('click');
  });

  it('renders the BaseDataView component with sidebar sidebarIsOpen', () => {
    store.open = true;
    wrapper = mountWrap(<div><MemoryRouter><BaseDataView store={store}/></MemoryRouter></div>);
    expect(wrapper.find('AppBar').length === 1).toEqual(true);
    wrapper.find('IconButton').first().simulate('click');
  });
});
