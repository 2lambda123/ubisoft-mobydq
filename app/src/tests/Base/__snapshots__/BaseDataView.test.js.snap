// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDataView component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(withRouter(Connect(BaseDataView))) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "appBar": "withRouter-Connect-BaseDataView---appBar-4",
        "appBarShift": "withRouter-Connect-BaseDataView---appBarShift-5",
        "appBarSpacer": "withRouter-Connect-BaseDataView---appBarSpacer-11",
        "chartContainer": "withRouter-Connect-BaseDataView---chartContainer-13",
        "content": "withRouter-Connect-BaseDataView---content-12",
        "drawerPaper": "withRouter-Connect-BaseDataView---drawerPaper-9",
        "drawerPaperClose": "withRouter-Connect-BaseDataView---drawerPaperClose-10",
        "formContainer": "withRouter-Connect-BaseDataView---formContainer-14",
        "menuButton": "withRouter-Connect-BaseDataView---menuButton-6",
        "menuButtonHidden": "withRouter-Connect-BaseDataView---menuButtonHidden-7",
        "root": "withRouter-Connect-BaseDataView---root-1",
        "tableContainer": "withRouter-Connect-BaseDataView---tableContainer-15",
        "title": "withRouter-Connect-BaseDataView---title-8",
        "toolbar": "withRouter-Connect-BaseDataView---toolbar-2",
        "toolbarIcon": "withRouter-Connect-BaseDataView---toolbarIcon-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "appBar": "withRouter-Connect-BaseDataView---appBar-4",
          "appBarShift": "withRouter-Connect-BaseDataView---appBarShift-5",
          "appBarSpacer": "withRouter-Connect-BaseDataView---appBarSpacer-11",
          "chartContainer": "withRouter-Connect-BaseDataView---chartContainer-13",
          "content": "withRouter-Connect-BaseDataView---content-12",
          "drawerPaper": "withRouter-Connect-BaseDataView---drawerPaper-9",
          "drawerPaperClose": "withRouter-Connect-BaseDataView---drawerPaperClose-10",
          "formContainer": "withRouter-Connect-BaseDataView---formContainer-14",
          "menuButton": "withRouter-Connect-BaseDataView---menuButton-6",
          "menuButtonHidden": "withRouter-Connect-BaseDataView---menuButtonHidden-7",
          "root": "withRouter-Connect-BaseDataView---root-1",
          "tableContainer": "withRouter-Connect-BaseDataView---tableContainer-15",
          "title": "withRouter-Connect-BaseDataView---title-8",
          "toolbar": "withRouter-Connect-BaseDataView---toolbar-2",
          "toolbarIcon": "withRouter-Connect-BaseDataView---toolbarIcon-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
